service/controller/azurecluster/handler/azureclusteridentity/create.go

```diff
@@ -99,6 +100,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	// Check if the AzureCluster CR has a SubscriptionID set.
+	if azureCluster.Spec.SubscriptionID == "" {
```

Comment:
In next PR's when we will use this field, we'd need to cancel reconciliation until this field is set.